packages/javascript/src/dispatcher.ts

```diff
@@ -0,0 +1,68 @@
+import { Queue } from "./queue"
+import { PushApi } from "./api"
+
+type DispatcherOptions = { limit: number; initialDuration: number }
+
+export class Dispatcher {
+  public options: DispatcherOptions
+
+  private _queue: Queue
+  private _api: PushApi
+  private _retries = 0
+  private _timerID = 0
+  private _duration = 0
+
+  constructor(
+    queue: Queue,
+    api: PushApi,
+    options?: Partial<DispatcherOptions>
+  ) {
+    this._api = api
+    this._queue = queue
+
+    this.options = {
+      limit: 5,
+      initialDuration: 1000,
+      ...options
+    }
+
+    this.reset()
+  }
+
+  public schedule(time = this._duration): number {
+    const cb = async () => {
```

Comment:
What does `cb` stand for in this case?

Reply:
`cb` === `callback`

Reply:
Ok :+1: No reason to change this now, but for next time it might be a good idea to call the variable callback, it makes things a bit more explicit.